lgc/state/Compiler.cpp

```diff
@@ -217,24 +220,29 @@ void PipelineState::generateWithNewPassManager(std::unique_ptr<Module> pipelineM
     return PipelineStateWrapper(getLgcContext());
   });
 
-  if (m_emitLgc)
+  if (m_emitLgc) {
     // -emit-lgc: Just write the module.
     passMgr->addPass(PrintModulePass(outStream));
-  else {
+    // Run the "whole pipeline" passes.
+    passMgr->run(*pipelineModule);
+  } else {
     // Patching.
-    Patch::addPasses(this, *passMgr, m_noReplayer, patchTimer, optTimer, std::move(checkShaderCacheFunc),
+    Patch::addPasses(this, *passMgr, !m_noReplayer, patchTimer, optTimer, std::move(checkShaderCacheFunc),
```

Comment:
Does adding the negation to `m_noReplayer ` change the behaviour?

Reply:
@piotrAMD so yes, this was a mistake I made in a previous patch. the `addPasses` function takes a `addReplayerPass` boolean, so we need the negation here.

Reply:
Ah I see, thanks.